pandas/tests/indexes/multi/test_duplicates.py

```diff
@@ -306,3 +306,20 @@ def test_duplicated_drop_duplicates():
     assert duplicated.dtype == bool
     expected = MultiIndex.from_arrays(([2, 3, 2, 3], [1, 1, 2, 2]))
     tm.assert_index_equal(idx.drop_duplicates(keep=False), expected)
+
+
+def test_multi_drop_duplicates_pos_args_deprecation():
+    idx = MultiIndex.from_arrays([[1, 2, 3, 1], [1, 2, 3, 1]])
+
+    msg = (
+        "In a future version of pandas all arguments of "
+        "Index.drop_duplicates will be keyword-only"
+    )
```

Comment:
The error message shows `Index` - to get it to show `MultiIndex`, you'll need to define `drop_duplicates` in the `MultiIndex` class and then call `super().drop_duplicates` inside it - see `interpolate` for an example of this

Reply:
Added this change in the latest commit.

I will have to do the same for #41551, as the test for MultiIndex currently has the same problem.